### PR TITLE
"Fix" and polish the Quilt Screen API

### PR DESCRIPTION
**Pull request by EnnuiL**
This PR attempts to clean up and do a tiny refactor on where it made sense on the Screen API (can't have breaking changes if the last few QSL releases are broken anyway), and also finally fixes a crash involving its injected interface; Unfortunately, the "fix" was to temporarily disable the injection until the refmap issues are no more

### CODE REVIEW

**Review by maximumpower55 — APPROVED** [review]
LGTM

**Review by PepperCode1 — APPROVED** [review]

File: library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/QuiltScreen.java [hunk 1]
```diff
@@ -28,37 +28,40 @@
...
-@InjectedInterface(Screen.class)
-public interface QuiltScreenExtensions {
+public interface QuiltScreen {
 	/**
-	 * Gets all of a screen's button widgets.
-	 * <p>
-	 * The provided list allows for addition and removal of buttons from the screen.
+	 * Gets all of the screen's button widgets.
+	 *
+	 * <p>The provided list allows for addition and removal of buttons from the screen.
```
Comment (LambdAurora) [inline library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/QuiltScreen.java]: Please don't move those `<p>` tags to a start of a paragraph in an inline way.

**Review by LambdAurora — CHANGES_REQUESTED** [review]

File: library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/QuiltScreen.java (hunk 1 above)
Comment (EnnuiL) [reply library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/QuiltScreen.java]: The Checkstyle config should be changed in order to reflect that then

Comment (EnnuiL) [reply library/gui/screen/src/main/java/org/quiltmc/qsl/screen/api/client/QuiltScreen.java]: I've changed it back so it followed CONTRIBUTING.md; This resulted into some of the FAPI copy-paste being quiltified in the process

**Review by LambdAurora — APPROVED** [review]

**Review by TheGlitch76 — APPROVED** [review]

**Pull request merged**